client/src/Components/Tabs.js

```diff
@@ -2,60 +2,85 @@ import React, { useState, useEffect } from "react";
 import Pans from "./Pans";
 
 const Tabs = (props) => {
-    const [data, setData] = useState(props.data);
-    const todayDate = new Date().getDate();
-    const [isToday, setIsToday] = useState(true);
-    const [isTmr, setIsTmr] = useState(false);
-    const [isLater, setIsLater] = useState(false);
-    const todayData = data.filter(
-      (ele) => new Date(ele.by_date).getDate() === todayDate
-    );
-    const tmrData = data.filter(
-      (ele) => new Date(ele.by_date).getDate() === todayDate + 1
-    );
-    const laterData = data.filter(
-      (ele) =>
-        new Date(ele.by_date).getDate() !== todayDate + 1
-        && new Date(ele.by_date).getDate() !== todayDate
-    );
-    useEffect(() => {
-        setData(props.data);
-    }, [props.data]);
+  const [data, setData] = useState(props.data);
+  const todayDate = new Date().getDate();
+  const [isToday, setIsToday] = useState(true);
```

Comment:
Could also batch the isToday, isTomorrow, isLater state into an object. So instead of calling setToday,  setTmr, setLater. Could just call setTimes({true, false, true})